developer_manual/digging_deeper/reference.rst

```diff
@@ -0,0 +1,527 @@
+you need to make sure the ``OCP\Collaboration\Reference\RenderReferenceEvent`` is dispatched
+before you load the page where you want to display link previews.
+
+For example, you can place this before returning your TemplateResponse in your controller:
+
+.. code-block:: php
+
+    $this->eventDispatcher->dispatchTyped(new OCP\Collaboration\Reference\RenderReferenceEvent());
+
+This is done in Text and Talk if you need more examples.
+
+On the frontend side, there are 3 ways to display link previews (also named reference widgets):
+
+* use the NcRichText Vue component
+* use the NcReferenceWidget Vue component
+* access the reference API directly and manually render the previews
+
+NcRichText
+~~~~~~~~~~
+
+Link previews will be automatically rendered for links in the content of the ``<NcRichText>`` Vue component.
+This component will take care of resolving the links itself.
+
+.. code-block:: html
+
+    <NcRichText :text="message"
+		:arguments="richParameters"
+		:autolink="true"
+		:reference-limit="0" />
+
+NcRichText can be imported like this:
+
+.. code-block:: javascript
+
+    import NcRichText from '@nextcloud/vue/dist/Components/NcRichText.js'
+
+
+`NcRichText component documentation <https://nextcloud-vue-components.netlify.app/#/Components/NcRichText?id=ncrichtext-1>`_
+
+NcReferenceWidget
+~~~~~~~~~~~~~~~~~
+
+You can display a preview for a specific link by using the ``<NcReferenceWidget>`` component.
+You need to ask the server to resolve the link to get a reference object that you can then give as a property
+to NcReferenceWidget.
+
+To resolve a link:
+
+.. code-block:: javascript
+
+    const myLink = 'https://github.com'
+    const requestOptions = {
+        params: {
+            reference: myLink,
+        },
+    }
+    axios.get(generateOcsUrl('references/resolve', 2), requestOptions)
+        .then((response) => {
+            reference = response.data.ocs.data.references[myLink]
+        })
+
+Then you can use the reference object you got:
+
+.. code-block:: html
+
+    <NcReferenceWidget :reference="reference" />
+
+API to resolve links
+~~~~~~~~~~~~~~~~~~~~
+
+Accessing the API directly can be useful if you want to:
+
+* resolve links from outside Nextcloud, in a client for example
+* manually resolve and render links instead of using the Vue components
+
+Endpoints to resolve links:
+
+* GET /ocs/v2.php/references/resolve
+    * ``reference`` parameter which is the link to resolve
+* GET /ocs/v2.php/references/resolve
+    * ``references`` parameter which is an array of links to resolve
+    * ``limit`` parameter which is the maximum number of links to resolve
+
+Request examples
+^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    curl -u USER:PASSWD -H "Accept: application/json" -H "ocs-apirequest: true" \
+        "https://my.nextcloud.org/ocs/v2.php/references/resolve?reference=https://github.com"
+
+will return an OCS response with that data:
+
+.. code-block:: json
+
```

Comment:
missing an opening { to become valid json

Reply:
It is just part of the json object. But yeah, let's mention the entire OCS response so it's valid json. :+1: